### ISSUE TITLE
Use VM definition or propTypes info, so that you can exclude properties option

### ISSUE THREAD

**Issue by BigAB** [canjs/react-view-models#18]
Just to make it a little of a nicer API, it would be great if:
 - any defined, non-method viewModel properties got read and passed into props
 - any lists read in were converted to arrays if the definition had an `asArray` property
 - any methods on the viewModel got bound in the constructor and copied into props on update
 - if the `spread` option is `true` also copy any props passed into the HOC
Failing that maybe if:
 - the child component (usually a presentational component) has its `propTypes` declared (a best practice in React anyway)
 - there is no `properties` property in the in options object

...then use the propTypes to determine the properties.

Basically match keys, but also if it's expecting an array, and we pass a list, use the `isArray` option.

**Comment by BigAB** [comment]
There is a branch `define-props-from-vm-definitions` that does this but tests are currently failing due to to changes from the observe-render branch that is in the branch.

**Comment by christopherjbaker** [comment]
closing as no-longer-relevant
